llvm/include/llvm/IR/PatternMatch.h

```diff
@@ -1711,6 +1711,30 @@ m_BitCast(const OpTy &Op) {
   return CastOperator_match<OpTy, Instruction::BitCast>(Op);
 }
 
+template <typename Op_t> struct ElementWiseBitCast_match {
+  Op_t Op;
+
+  ElementWiseBitCast_match(const Op_t &OpMatch) : Op(OpMatch) {}
+
+  template <typename OpTy> bool match(OpTy *V) {
+    if (auto *I = dyn_cast<BitCastInst>(V)) {
+      Type *SrcType = I->getSrcTy();
+      Type *DstType = I->getType();
+      // Make sure the bitcast doesn't change between scalar and vector and
+      // doesn't change the number of vector elements.
+      if (SrcType->isVectorTy() == DstType->isVectorTy() &&
+          SrcType->getScalarSizeInBits() == DstType->getScalarSizeInBits())
```

Comment:
I think the cleaner way to express this would be that the element count is the same. It comes down to the same thing, but note that getScalarSizeInBits() is ill-defined for pointers -- you do get away with this here because currently only trivial pointer bitcasts are allowed.

That should also allow you to find some more instances of this pattern, e.g. in FoldOpIntoSelect.

Reply:
Done.